starfish/util/logging.py

```diff
@@ -0,0 +1,51 @@
+import platform
+from functools import lru_cache
+from json import JSONEncoder
+from subprocess import (
+    CalledProcessError,
+    check_output,
+)
+
+import pkg_resources
+
+import starfish
+from starfish.types import CORE_DEPENDENCIES
+
+
+@lru_cache(maxsize=1)
+def get_core_dependency_info():
+    dependency_info = dict()
+    for dependency in CORE_DEPENDENCIES:
+        version = pkg_resources.get_distribution(dependency).version
+        dependency_info[dependency] = version
+    return dependency_info
+
+
+@lru_cache(maxsize=1)
+def get_git_commit_hash():
+    # First check if in starfish repo
+    try:
+        check_output(["git", "ls-files", "--error-unmatch", starfish.__file__])
+        return check_output(["git", "describe", "--always"]).strip()
```

Comment:
This should not be in the try-block.  It would seem that a failure here is indicative of a bug, rather than starfish is not under git tracking.

Reply:
Are you talking about the `git describe` or the `git ls-files --error-unmatch`? Because the second is just list the files under tracking and error if starfish.__file__ is not one of them, isn't that the behavior we want?

Reply:
I am pretty sure that all you need to do is to execute `git ls-files os.path.basename(starfish.__file__)`, after you chdir to `os.path.dirname(starfish.__file__)`.

git ls-files of a file tracked by git, but not while you are in the directory fails:
```
[tt]:~> git ls-files microscopy/starfish-git/starfish/__init__.py
fatal: not a git repository (or any of the parent directories): .git
```

but chdir and ls-files and it works:

```
[tt]:~> cd microscopy/starfish-git/starfish/
[tt]:~/microscopy/starfish-git/starfish:tonytung-pr-684> git ls-files __init__.py
__init__.py
[tt]:~/microscopy/starfish-git/starfish:tonytung-pr-684> 
```